utils/python/CIME/SystemTests/tests/test_system_tests_compare_two.py

```diff
@@ -0,0 +1,384 @@
+#!/usr/bin/env python
```

Comment:
Note that I have unit tests living in a tests directory that sits along-side the code being tested. I find this to be an easy-to-work-with organization of test code, but am open to reorganizing this if others prefer some different organization.

One thing I'd like to avoid is having test code living in the same module as the production code: I feel like that makes it harder to understand what's test code and what's production code.